src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -787,6 +807,47 @@ SPECTRE_ALWAYS_INLINE const auto& get(const DataBox<TagList>& box) {
   return box.template get<Tag>();
 }
 
+////////////////////////////////////////////////////////////////
+// Copy mutable creation items from the DataBox
+
+/// \cond
+template <typename... Tags>
+template <typename Tag>
+SPECTRE_ALWAYS_INLINE auto DataBox<tmpl::list<Tags...>>::copy_item() const {
+  using item_tag = detail::first_matching_tag<tags_list, Tag>;
+  using item_type = typename item_tag::type;
+  static_assert(tmpl::list_contains_v<mutable_item_creation_tags, item_tag>,
+                "Can only copy mutable creation items");
+  return deserialize<item_type>(
+      serialize<item_type>(get_item<item_tag>().get()).data());
+}
+
+template <typename... DbTags>
+template <typename... CopiedItemsTags>
+tuples::TaggedTuple<CopiedItemsTags...>
+DataBox<tmpl::list<DbTags...>>::copy_items(
+    tmpl::list<CopiedItemsTags...> /*meta*/) const {
+  return tuples::TaggedTuple<CopiedItemsTags...>{
+      copy_item<CopiedItemsTags>()...};
+}
+/// \endcond
+
+/*!
+ * \ingroup DataBoxGroup
+ * \brief Copy the items from the DataBox into a TaggedTuple
+ *
+ * \return The objects corresponding to CopiedItemsTagList
+ *
+ * \note The tags in CopiedItemsTagList must be a subset of
+ * the mutable_item_creation_tags of the DataBox
+ */
+template <typename CopiedItemsTagList, typename DbTagList>
+SPECTRE_ALWAYS_INLINE auto copy_items(const DataBox<DbTagList>& box) {
```

Comment:
do you only want a copy, or would a move one be useful too? If it's always a copy, that's fine, just asking :)

Reply:
A move could also be useful and might be the right choice; I'll do that in a separate PR as moving an item will need to reset subitems and compute items that depend on the moved item...